src/ai.js

```diff
@@ -18,11 +19,790 @@ function getInternalGame() {
+            Object.entries(p).forEach(e => {
+                let [key, val] = e;
+                if (typeof val !== "number") return;
+                if (val == 0) return;
+                if (["id"].includes(key)) return;
+
+                if (["fatigue", "heroPowerCost", "overload"].includes(key)) val = -val;
+
+                let bias = 1;
+                if (p.id != this.plr.id) bias = -1;
+
+                score += val * bias * VALUE_BIAS;
+            });
+        });
+
+        [simulation.player1, simulation.player2].forEach(p => {
+            [p.deck.length, p.hand.length, p.quests.length, p.sidequests.length, p.secrets.length].forEach(val => {
+                let bias = 1;
+                if (p.id != this.plr.id) bias = -1;
+
+                score += val * bias * VALUE_BIAS;
+            });
+        });
+
+        game.log(`AI: Evaluation completed with a score of ${score.toPrecision(2)}`, !this.logAction);
+
+        return score;
+    }
+
+    /**
+     * Evaluates a card
+     * 
+     * @param {Card} c The card
+     * @param {boolean} [sentiment=true] If it should perform sentiment analysis on the card's description
+     * 
+     * @returns {number} The score
+     */
+    _evaluateCard(c, sentiment = true) {
+        let score = 0;
+
+        if (c.type == "Minion" || c.type == "Weapon") score += (c.getAttack() + c.getHealth()) * game.config.AIStatsBias;
+        else score += game.config.AISpellValue * game.config.AIStatsBias; // If the spell value is 4 then it the same value as a 2/2 minion
+        score -= c.mana * game.config.AIManaBias;
+
+        c.keywords.forEach(() => score += game.config.AIKeywordValue);
+        Object.values(c).forEach(c => {
+            if (c instanceof Array && c[0] instanceof Function) score += game.config.AIFunctionValue;
+        });
+
+        if (sentiment) score += this.backup_ai.analyzePositive(c.desc || "");
+        if (!c.desc) score -= game.config.AINoDescPenalty;
+
+        return score;
+    }
+
+    /**
+     * Creates and returns a simulation
+     * 
+     * @returns {Game} The simulation
+     */
+    _createSimulation() {
+        // Make a deep copy of the current game
+        // Don't copy these props
+        // FIXME: Why. Gets caught in an infinite loop.
+        // FIXME: Mana increases without turn ending.
+        // FIXME: The ai cannot play cards.
+        game.log("AI: Creating simulation...", !this.logAction);
+
+        let count = 0;
+
+        /**
+         * @type {Game}
+         */
+        let simulation = lodash.cloneDeepWith(game, ()=>{count++});
+        game.log(`AI: Cloned total ${count} nodes from rows`, !this.logAction);
+        simulation.simulation = true; // Mark it as a simulation
+
+        [...simulation.player1.deck, ...simulation.player1.hand, ...simulation.player2.deck, ...simulation.player2.hand, ...simulation.player.deck, ...simulation.player.hand, ...simulation.opponent.deck, ...simulation.opponent.hand].forEach(c => {
```

Comment:
Aren't these the same thing? Isn't `game.player` just a reference to `game.player1` or maybe cloneDeep changes that?